common/browser_context.go

```diff
@@ -232,28 +235,22 @@ func (b *BrowserContext) GrantPermissions(permissions []string, opts goja.Value)
 		"clipboard-write":      cdpbrowser.PermissionTypeClipboardSanitizedWrite,
 		"payment-handler":      cdpbrowser.PermissionTypePaymentHandler,
 	}
-	origin := ""
-
-	rt := b.vu.Runtime()
-	if opts != nil && !goja.IsUndefined(opts) && !goja.IsNull(opts) {
-		opts := opts.ToObject(rt)
-		for _, k := range opts.Keys() {
-			if k == "origin" {
-				origin = opts.Get(k).String()
-				break
-			}
-		}
-	}
 
 	perms := make([]cdpbrowser.PermissionType, 0, len(permissions))
 	for _, p := range permissions {
-		perms = append(perms, permsToProtocol[p])
+		proto, ok := permsToProtocol[p]
+		if !ok {
+			return fmt.Errorf("%q is an invalid permission", p)
+		}
+		perms = append(perms, proto)
 	}
 
-	action := cdpbrowser.GrantPermissions(perms).WithOrigin(origin).WithBrowserContextID(b.id)
+	action := cdpbrowser.GrantPermissions(perms).WithOrigin(parsedOpts.Origin).WithBrowserContextID(b.id)
 	if err := action.Do(cdp.WithExecutor(b.ctx, b.browser.conn)); err != nil {
-		k6ext.Panic(b.ctx, "internal error while granting browser permissions: %w", err)
+		return fmt.Errorf("error granting browser permissions: %w", err)
```

Comment:
Should we follow our typical error text also here without saying "error"?

```suggestion
		return fmt.Errorf("granting browser permissions: %w", err)
```